### ISSUE TITLE
[FEATURE]: Add Support for Maximum Application Attempts Per Company

### ISSUE THREAD

**Issue by moudimash99** [feder-cr/Jobs_Applier_AI_Agent#935]
### Feature summary

Introduce a configurable maximum number of applications per company, allowing users to specify limits like 5 applications to Google instead of just a boolean option.

### Feature description

Currently, the bot offers a binary option (apply_once_at_company: true/false) to either apply once or indefinitely to the same company. This proposed feature introduces a more flexible system where users can define a specific maximum number of applications per company. For example, users could configure the bot to apply to Google up to 5 times but still apply indefinitely to other companies.

This feature would involve:

Updating the work_preferences.yaml file to accept numeric values for apply_once_at_company.
Implementing logic to track and limit applications based on the specified number.

### Motivation

Provides users with better control over application strategies by avoiding over-applying to specific companies.

**Comment by moudimash99** [comment]
id like to be assigneed to this

**Comment by surapuramakhil** [comment]
Hey @moudimash99 , can you also take this https://github.com/AIHawk-FOSS/Auto_Jobs_Applier_AI_Agent/issues/832 as both are closely related.

**Comment by moudimash99** [comment]
Yeah I think that'll be the next thing I'm going to take, with them both moving forward. We'll need surely a time limit as applying to a company multiple times is only relevant in a time frame. I was thinking of using the current architecture,are there other features that are or will be using a DB as that might be interesting moving forward for even other features

**Comment by surapuramakhil** [comment]
@cjbbb @moudimash99 can you work together on this and #832, both are highly coupled - it's easy both doing at once

**Comment by moudimash99** [comment]
on it chief. @surapuramakhil 

@cjbbb wanna have a quick chat sometime about it? it's a simple thing but we can halve the work or so

**Comment by cjbbb** [comment]
> on it chief. @surapuramakhil
> 
> @cjbbb wanna have a quick chat sometime about it? it's a simple thing but we can halve the work or so@cjbbb想找个时间快速聊聊这件事吗？这是一件简单的事情，但我们可以将工作量减半左右

hi @moudimash99 what is your discord account, we can chat on it

**Comment by moudimash99** [comment]
@cjbbb moudimash99
